src/modules/fundingManager/oracle/LM_ManualExternalPriceSetter_v1.sol

```diff
@@ -0,0 +1,181 @@
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to
+ *                          our Security Policy at security.inverter.network or
+ *                          email us directly!
+ *
+ * @author  Zealynx Security
+ */
+contract LM_ManualExternalPriceSetter_v1 is
+    ILM_ManualExternalPriceSetter_v1,
+    IOraclePrice_v1,
+    Module_v1
+{
+    //--------------------------------------------------------------------------
+    // Constants
+
+    bytes32 public constant PRICE_SETTER_ROLE = "PRICE_SETTER_ROLE";
+    uint8 private constant INTERNAL_DECIMALS = 18;
+
+    //--------------------------------------------------------------------------
+    // State Variables
+
+    /// @notice The price for issuing tokens (normalized to INTERNAL_DECIMALS)
+    uint256 private _issuancePrice;
+
+    /// @notice The price for redeeming tokens (normalized to INTERNAL_DECIMALS)
+    uint256 private _redemptionPrice;
+
+    /// @notice Decimals of the input token
+    uint8 private _inputTokenDecimals;
+
+    /// @notice Decimals of the output token
+    uint8 private _outputTokenDecimals;
+
+    /// @dev Storage gap for upgradeable contracts
+    uint[50] private __gap;
+
+    //--------------------------------------------------------------------------
+    // Initialization
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata_,
+        bytes memory configData_
+    ) external override(Module_v1) initializer {
+        __Module_init(orchestrator_, metadata_);
+
+        // Decode input and output token addresses from configData_
+        (address inputToken, address outputToken) = abi.decode(
+            configData_,
+            (address, address)
+        );
+
+        // Store token decimals for price normalization
+        _inputTokenDecimals = IERC20Metadata(inputToken).decimals();
+        _outputTokenDecimals = IERC20Metadata(outputToken).decimals();
+    }
+
+    //--------------------------------------------------------------------------
+    // Interface Support
+
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        override
+        returns (bool)
+    {
+        return interfaceId == type(ILM_ManualExternalPriceSetter_v1).interfaceId
+            || super.supportsInterface(interfaceId);
+    }
+
+    //--------------------------------------------------------------------------
+    // External Functions
+
+    /// @inheritdoc ILM_ManualExternalPriceSetter_v1
+    function setIssuancePrice(uint256 price_)
+        external
+        onlyModuleRole(PRICE_SETTER_ROLE)
+    {
+        if (price_ == 0) revert Module__LM_ExternalPriceSetter__InvalidPrice();
+        
+        // Normalize price to internal decimal precision
+        _issuancePrice = _normalizePrice(price_, _inputTokenDecimals);
+        emit PriceSet(price_);
```

Comment:
We prob only need one of the two emits. Not sure if I commented on that already, making you implement it in this way

Reply:
removed `PriceSet`